tests/unit/ReferenceListTest.php

```diff
@@ -372,14 +372,4 @@ public function testGivenNonEmptyList_isNotEmpty() {
 		$this->assertFalse( $references->isEmpty() );
 	}
 
-	public function testGivenNonEmptyListWithForwardedIterator_isNotEmpty() {
-		$references = new ReferenceList();
-		$references->addNewReference( new PropertyNoValueSnak( 1 ) );
-		$references->next();
```

Comment:
Code typically does not type hint against SplObjectStorage or Countable or Traversable. Sure, all this should be mentioned in the release notes, but is barely helpful.

"Oh, wow" was meant as "oh, wow, PHP is broken". I reviewed the changes that touch this code a long, long time ago and I'm looking at them again for almost a full day now, and I still don't fully understand what's going on. There is a reason I never merged these patches. Verbose, crystal clear documentation is absolutely crucial here.